### PR TITLE
feat: Add Bubble chart migration logic

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
This PR adds the Bubble chart migration logic (NVD3 ➡️ ECharts). Users can execute this migration using the https://github.com/apache/superset/pull/25304 CLI command and disable the legacy version with the [VIZ_TYPE_DENYLIST](https://github.com/apache/superset/blob/ec6191023263ac5f8292dd37f037805c3196e029/superset/config.py#L829) configuration.

@sadpandajoe @jinghua-qa

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screenshot 2023-11-20 at 11 25 25
Screenshot 2023-11-20 at 11 29 29

### TESTING INSTRUCTIONS
1 - Upgrade a Bubble chart using the CLI command
2 - Check the new chart
3 - Downgrade a Bubble chart using the CLI command
4 - Check the legacy chart

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/migrations/shared/migrate_viz/processors.py [hunk 1]
```diff
@@ -184,3 +184,32 @@ def _pre_action(self) -> None:
         )
 
         self.data["opacity"] = 0.7
+
+
+class MigrateBubbleChart(MigrateViz):
+    source_viz_type = "bubble"
+    target_viz_type = "bubble_v2"
+    rename_keys = {
+        "bottom_margin": "x_axis_title_margin",
```
Comment (john-bodley) [inline superset/migrations/shared/migrate_viz/processors.py:193]: Unrelated, but what's the reason we have a mix of snake- and camel-case in the chart properties?

File: superset/migrations/shared/migrate_viz/processors.py [hunk 2]
```diff
@@ -184,3 +184,32 @@ def _pre_action(self) -> None:
...
+        bottom_margin = self.data.get("bottom_margin")
+        if self.data.get("x_axis_label") and (
+            not bottom_margin or bottom_margin == "auto"
+        ):
+            self.data["bottom_margin"] = 30
+
+        if x_ticks_layout := self.data.get("x_ticks_layout"):
+            self.data["x_ticks_layout"] = 45 if x_ticks_layout == "45°" else 0
+
+        # Truncate Y-axis by default to preserve layout
```
Comment (john-bodley) [inline superset/migrations/shared/migrate_viz/processors.py:214]:
```suggestion
        # Truncate y-axis by default to preserve layout
```

File: superset/migrations/shared/migrate_viz/processors.py [hunk 3]
```diff
@@ -184,3 +184,32 @@ def _pre_action(self) -> None:
...
+        self.data["y_axis_showminmax"] = True
```
Comment (john-bodley) [inline superset/migrations/shared/migrate_viz/processors.py:215]: Just to confirm, this occurs prior to the renaming of the keys?

File: superset/migrations/shared/migrate_viz/processors.py [hunk 4]
```diff
@@ -184,3 +184,32 @@ def _pre_action(self) -> None:
...
+        "y_log_scale": "logYAxis",
+    }
+    remove_keys = {"x_axis_showminmax"}
+
+    def _pre_action(self) -> None:
+        bottom_margin = self.data.get("bottom_margin")
+        if self.data.get("x_axis_label") and (
+            not bottom_margin or bottom_margin == "auto"
+        ):
+            self.data["bottom_margin"] = 30
```
Comment (john-bodley) [inline superset/migrations/shared/migrate_viz/processors.py:209]: Any reason the bottom margin is now fix to 30 (pixels?) as opposed to being automatic?

File: superset/migrations/shared/migrate_viz/processors.py (hunk 3 above)
Comment (michael-s-molina) [reply superset/migrations/shared/migrate_viz/processors.py:215]: Yep

File: superset/migrations/shared/migrate_viz/processors.py (hunk 4 above)
Comment (michael-s-molina) [reply superset/migrations/shared/migrate_viz/processors.py:209]: Layout adjustments to match the legacy version look.

File: superset/migrations/shared/migrate_viz/processors.py (hunk 1 above)
Comment (michael-s-molina) [reply superset/migrations/shared/migrate_viz/processors.py:193]: 😏

**Review by john-bodley — APPROVED** [review]
LGTM

**Pull request merged**